tls/s2n_connection.c

```diff
@@ -528,6 +527,62 @@ void *s2n_connection_get_ctx(struct s2n_connection *conn)
     return conn->context;
 }
 
+int s2n_connection_release_buffers(struct s2n_connection *conn)
+{
+    GUARD(s2n_connection_free_handshake(conn));
+
+    /* wipe and truncate the input and output buffers while we are idle */
+    GUARD(s2n_stuffer_wipe(&conn->in));
+    GUARD(s2n_stuffer_wipe(&conn->out));
+
+    GUARD(s2n_stuffer_resize(&conn->in, 0));
+    GUARD(s2n_stuffer_resize(&conn->out, 0));
+
+    return 0;
+}
+
+int s2n_connection_realloc_buffers(struct s2n_connection *conn)
+{
+    GUARD(s2n_stuffer_resize(&conn->in, S2N_LARGE_FRAGMENT_LENGTH));
+    GUARD(s2n_stuffer_resize(&conn->out, S2N_LARGE_RECORD_LENGTH));
+
+    return 0;
+}
+
+int s2n_connection_free_handshake(struct s2n_connection *conn)
+{
+    /* We are done with the handshake */
+    GUARD(s2n_hash_reset(&conn->handshake.md5));
```

Comment:
Can we release these immediately after handshake completion?

Reply:
possibly - ideally we would release things we don't need as soon as we are done with them.  the only concern i have there is whether that will interfere with something we might add, such as renegotiation, in the future.  however releasing them inline without the separate API would at least catch that failure during development, rather than relying on the correct unit test being written which uses the wipe API.

Reply:
```
Running s2n_mem_usage_test.c                               ...
VmData initial:               819200
VmData after allocations:   31510528
VmData after handshakes:    49057792
VmData after free:          49057792
VmData after release:       31588352
Max VmData diff allowed:    56320000
```

given this, i have added expectations that free doesn't add memory and release gets us to within 1K per connection of the initial alloced state.